src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -2912,7 +2920,7 @@ private void InitSetter()
                             errType = typeof(object);
                         }
                     }
-                    else if (readOnly)
+                    else if (readOnly || initOnly)
```

Comment:
I feel there should be a specific error message for the init-only case.

Reply:
I thought about it too, but decided against. 

The caller doesn't need to know or care whether the property was "initializable" at some point in the past, all that matters is that it isn't settable _right now_. 

If we surface this distinction in the error messages, then I think we should surface the distinction in the rest of the type-system as well, eg. display hints like `{ get; init; }` instead of `{ get; }`, perhaps add an `IsInitOnly` property to `PSProperty`.

I'm happy to do it either way, I just think it should be consistent :)

Reply:
I see your point and agree.